### PR TITLE
Added Bozorgnia & Campbell (2016) GMPE

### PR DESCRIPTION
**Pull request by ftbernales**
Implements the BC15 GMPE by Bozorgnia & Campbell (2016) for vertical-component ground motions from the PEER NGA-West2 Project (3 of 4).

This is largely similar in form to the CB14 horizontal-component model by Campbell & Bozorgnia (2014), so I called some methods therein, as needed.

I used the Excel file published together with the Earthquake Spectra paper for code verification. I detected an error in one of the formulas (as commented in the test .py file), and added VBA codes to generate the .csv files in the test suite. I shall send to you these files in a separate email for your reference.

### CODE REVIEW

File: openquake/hazardlib/gsim/bozorgnia_campbell_2016.py [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+"""
+import numpy as np
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable
+from openquake.hazardlib.gsim.campbell_bozorgnia_2014 import \
+                                          CampbellBozorgnia2014 as CB14
+from openquake.hazardlib import const
+from openquake.hazardlib.imt import PGA, PGV, SA
+
+
+class BozorgniaCampbell2016(GMPE):
```
Comment (mmpagani) [inline openquake/hazardlib/gsim/bozorgnia_campbell_2016.py:35]: If the functional form is the same used in CB14 wouldn't it be better to create this new model by extending the original class as in this example:

```
class BozorgniaCampbell2016(CB14):
```
In this way, you can avoid rewriting many of the methods as you did further down.

**Review by mmpagani — COMMENTED** [review]
Other than the comment provided, I am fine with this PR.

File: openquake/hazardlib/gsim/bozorgnia_campbell_2016.py (hunk 1 above)
Comment (ftbernales) [reply openquake/hazardlib/gsim/bozorgnia_campbell_2016.py:35]: Thanks for the suggestion, Marco. I would also agree with you that extending the CB14 class would give a better-looking code. I would like to note, however, that my reason for not doing that is by extending CB14, the new `BozorgniaCampbell2016` class would also get the now unnecessary methods (in particular, as is the general case in vertical-component GMPEs, the nonlinear functions for site response). I personally don't like this idea of inheriting unnecessary methods from CB14, so I just sacrificed aesthetics instead.

That being said, I would like to ask you and @micheles if there is a way that we can achieve BOTH CASES, i.e. not inheriting unnecessary methods from CB14 AND not having too many method calls from CB14.

Would appreciate further suggestions in light of the considerations above. Thanks!

Comment (micheles) [reply openquake/hazardlib/gsim/bozorgnia_campbell_2016.py:35]: Clearly we can introduce a base class containing the essential methods and have both CB14 and BC15 inheriting from it, each one adding its own methods.

Comment (mmpagani) [reply openquake/hazardlib/gsim/bozorgnia_campbell_2016.py:35]: Hi Francis, thanks for your thoughts. I agree with your strategy. Let's keep it this way for the time being. Thank you again for your contributions.

Comment (mmpagani) [reply openquake/hazardlib/gsim/bozorgnia_campbell_2016.py:35]: LGTM

**Pull request merged**